topaz/extract.py

```diff
@@ -115,134 +248,320 @@ def extract_auprc(targets:np.ndarray, scores:np.ndarray, radius:float, threshold
+            print(f"\nProcessing tomogram: {path}")
+            start_time = time.time()
+            image = load_image(path, make_image=False, return_header=False)
+            original_shape = image.shape
+            print(f"Tomogram shape: {original_shape}")
+            
+            # Define patch size and overlap
+            patch_size = (32, 128, 128)  # (z, y, x)
+            overlap = (8, 32, 32)  # 25% overlap
+            
+            scores = np.zeros(original_shape, dtype=np.float32)
+            weights = np.zeros(original_shape, dtype=np.float32)
+            
+            total_patches = ((original_shape[0] - overlap[0]) // (patch_size[0] - overlap[0]) + 1) * \
+                            ((original_shape[1] - overlap[1]) // (patch_size[1] - overlap[1]) + 1) * \
+                            ((original_shape[2] - overlap[2]) // (patch_size[2] - overlap[2]) + 1)
+            
+            with tqdm(total=total_patches, desc="Processing patches", unit="patch") as pbar:
+                for z in range(0, original_shape[0] - overlap[0], patch_size[0] - overlap[0]):
+                    for y in range(0, original_shape[1] - overlap[1], patch_size[1] - overlap[1]):
+                        for x in range(0, original_shape[2] - overlap[2], patch_size[2] - overlap[2]):
+                            z_end = min(z + patch_size[0], original_shape[0])
+                            y_end = min(y + patch_size[1], original_shape[1])
+                            x_end = min(x + patch_size[2], original_shape[2])
+                            
+                            patch = image[z:z_end, y:y_end, x:x_end]
+                            patch = patch[None, None, ...]  # Add batch and channel dimensions
+                            
+                            patch_scores = process_patch(patch)
+                            
+                            # Apply Gaussian smoothing to the patch
+                            patch_scores = gaussian_filter(patch_scores[0], sigma=1)
+                            
+                            # Create a weight array for blending
+                            weight = np.ones_like(patch_scores)
+                            weight = gaussian_filter(weight, sigma=2)
+                            
+                            scores[z:z_end, y:y_end, x:x_end] += patch_scores * weight
+                            weights[z:z_end, y:y_end, x:x_end] += weight
+                            
+                            del patch
+                            if use_cuda:
+                                torch.cuda.empty_cache()
+                            pbar.update(1)
+            
+            # Normalize the scores
+            scores /= np.maximum(weights, 1e-10)
+            
+            end_time = time.time()
+            print(f"Tomogram processing completed in {end_time - start_time:.2f} seconds")
+            yield path, scores, original_shape
```

Comment:
Todo - make score patching optional (probably tied with extract patching) and push score patching into the `topaz.predict` module